python/pyspark/sql/streaming/listener.py

```diff
@@ -477,7 +477,8 @@ def fromJson(cls, j: Dict[str, Any]) -> "StreamingQueryProgress":
             name=j["name"],
             timestamp=j["timestamp"],
             batchId=j["batchId"],
-            batchDuration=j["batchDuration"],
+            # before spark 4.0, batchDuration is not in the json method of jvm side StreamingQueryProgress
+            batchDuration=j["batchDuration"] if "batchDuration" in j else None,
```

Comment:
@dongjoon-hyun Hi Dongjoon, sorry for the back and forth. On second thought I actually find out that the newly added tests and the test failure in https://github.com/apache/spark/pull/42521#issuecomment-1691547730 actually finds out a bug. Here before I assume `batchDuration` is always in the passed in json, but before 4.0 it is not there.

Given that we don't add the change in https://github.com/apache/spark/pull/42077, this check is needed. I reverted that commit, and add this check.